src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,124 @@
+                + "    |____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("    Welcome to \n" + logo + "\n    Your personal assistant :)");
+        showLine();
+        showLine();
+        return "    Welcome to \n" + logo + "\n    Your personal assistant :)" + showLine() + showLine();
+    }
+
+    /** Prints a breakLine. */
+    public String showLine() {
+        System.out.println("    ______________________________________________________");
+        return "    ______________________________________________________";
+    }
+
+    /** Prints loading error.
+     * @param error error message.
+     */
+    public String showLoadingError(String error) {
+        System.out.println(error);
+        return error;
+    }
+
+    /**
+     * Prints duke exception error message.
+     * @param error Duke exception message.
+     */
+    public String showError(String error) {
+        System.out.println(error);
+        return error;
+    }
+
+    /** Prints exit statement. */
+    public String showExit() {
+        System.out.println("    Bye. Hope to see you again soon!");
+        return "    Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Print task description.
+     * @param counter Index of the task in the list.
+     * @param task The task to print.
+     */
+    public String printTask(int counter, Task task) {
+        System.out.println("    " + counter + ": " + task.toString());
+        return "    " + counter + ": " + task.toString();
+    }
+
+    /**
+     * Prints the search result.
+     */
+    public String printResult() {
+        System.out.println("    Here are the matching tasks in your list:");
+        return "    Here are the matching tasks in your list:";
+    }
+
+    /**
+     * Prints the add task message.
+     * @param listSize The size of the task list.
+     * @param task The task to add into the list.
+     */
+    public String addTask(int listSize, Task task) {
+        System.out.println("    Got it. I've added this task: \n     " + task.toString());
+        System.out.println("    Now you have " + listSize + " tasks in the list.");
+        return "    Got it. I've added this task: \n     " + task.toString()
+                + "\n    Now you have " + listSize + " tasks in the list.";
+    }
+
+    /**
+     * Prints the done message.
+     * @param task The task that is completed.
+     */
+    public String markDone(Task task) {
+        System.out.println("    Nice! I've marked this task as done:");
+        System.out.println("    " + task.toString());
+        return "    Nice! I've marked this task as done:"
+                + "\n    " + task.toString();
+    }
+
+    /**
+     * Prints the delete message.
+     * @param listSize The size of the task list.
+     * @param task The task to delete from the list.
+     */
+    public String markDelete(int listSize, Task task) {
+        System.out.println("    Noted. I've removed this task:");

```

Comment:
I noticed you've hardcoded a lot of indentations to format the displays in your print messages, perhaps these could be abstracted out into a printMessage method in Ui that auto appends message to the spaces?